storm-server/src/test/java/org/apache/storm/scheduler/resource/strategies/scheduling/TestLargeCluster.java

```diff
@@ -0,0 +1,458 @@
+
+        } else {
+            // this non-uniform supervisor distribution closelt (but not exactly) mimics IridiumBlue cluster
+            int numSupersPerRack = 82;
+            int numPorts = 50;
+
+            Map<String, SupervisorDetails> retList = new HashMap<>();
+
+            for (int rack = 0 ; rack < 12 ; rack++) {
+                double cpu = 3600; // %percent
+                double mem = 178_000; // MB
+                for (int superInRack = 0; superInRack < numSupersPerRack ; superInRack++) {
+                    createAndAddOneSupervisor(rack, superInRack, cpu - 100 * (superInRack % 2), mem, numPorts, retList);
+                }
+            }
+            for (int rack = 12 ; rack < 14 ; rack++) {
+                double cpu = 2400; // %percent
+                double mem = 118_100; // MB
+                for (int superInRack = 0; superInRack < numSupersPerRack ; superInRack++) {
+                    createAndAddOneSupervisor(rack, superInRack, cpu - 100 * (superInRack % 2), mem, numPorts, retList);
+                }
+            }
+            for (int rack = 14 ; rack < 16 ; rack++) {
+                double cpu = 1200; // %percent
+                double mem = 42_480; // MB
+                for (int superInRack = 0; superInRack < numSupersPerRack ; superInRack++) {
+                    createAndAddOneSupervisor(rack, superInRack, cpu - 100 * (superInRack % 2), mem, numPorts, retList);
+                }
+            }
+            return retList;
+        }
+    }
+
+    /**
+     * Create a large cluster, read topologies and configuration from resource directory and schedule.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testLargeCluster() throws Exception {
+        boolean uniformSupervisors = false; // false means non-uniform supervisor distribution
+
+        Map<String, SupervisorDetails> supervisors = createSupervisors(uniformSupervisors);
+
+        TopologyDetails[] topoDetailsArray = createTopoDetailsArray(false);
+        Assert.assertTrue("No topologies found", topoDetailsArray.length > 0);
+        Topologies topologies = new Topologies(topoDetailsArray);
+
+        Config confWithDefaultStrategy = new Config();
+        confWithDefaultStrategy.putAll(topoDetailsArray[0].getConf());
+        confWithDefaultStrategy.put(Config.TOPOLOGY_SCHEDULER_STRATEGY, DefaultResourceAwareStrategy.class.getName());
+
+        INimbus iNimbus = new INimbusTest();
+        Cluster cluster = new Cluster(iNimbus, new ResourceMetrics(new StormMetricsRegistry()), supervisors, new HashMap<>(),
+                topologies, confWithDefaultStrategy);
+
+        scheduler = new ResourceAwareScheduler();
+
+        List<Class> classesToDebug = Arrays.asList(DefaultResourceAwareStrategy.class,
+                GenericResourceAwareStrategy.class, ResourceAwareScheduler.class,
+                Cluster.class // count calls to calculateSharedOffHeapNodeMemory()
```

Comment:
What does the comment mean here?

Reply:
I had debug statement earlier in Cluster.java - since removed. Comment is not relevant any more.
Removed.